app/controllers/comment_controller.php

```diff
@@ -63,26 +80,102 @@ public function edit()
+                throw new NotFoundException("{$page} not found");
+                break;
         }
 
         $this->set(get_defined_vars());
         $this->render($page);
     }
+
+    public function delete()
+    {
+        $comments = new Comment();
+
+        $comment_id = Param::get('comment_id');
+
+        $comments->delete($comment_id);
+
+        $this->set(get_defined_vars());
+        redirect("/comment/view?thread_id={$_SESSION['thread_id']}");
+    }
+
+    public function liked()
+    {
+        $comment = new Comment();
+
+        $comment_id = Param::get('comment_id');
+        $like_checker = $comment->hasLiked($comment_id);
+        $dislike_checker = $comment->hasDisliked($comment_id);
+
+        if (!$like_checker) { //comment has not been liked
+            if (!$dislike_checker) { //comment has not been disliked
+                $comment->addLike($comment_id); //add 1 liked record in like_monitor table then increment liked column in comment table
+            } else { //comment has been disliked
+                $comment->subtractDislikedCount($comment_id); //subtract 1 from disliked column in comment table
+                $comment->deleteExisting($comment_id); //deletes disliked record in like_monitor table
+                $comment->addLike($comment_id); //add 1 like record in like_monitor table then increment liked column in comment table
+            }
+        }
+
+        $this->set(get_defined_vars());
+        redirect("/comment/view?page={$_SESSION['current_page']}&thread_id={$_SESSION['thread_id']}");
+    }
+
+    public function disliked()
+    {
+        $comment = new Comment();
+
+        $comment_id = Param::get('comment_id');
+        $like_checker = $comment->hasLiked($comment_id);
+        $dislike_checker = $comment->hasDisliked($comment_id);
+
+        if (!$dislike_checker) { //comment has not been disliked
+            if (!$like_checker) { //comment has not been liked
+                $comment->addDislike($comment_id); //add 1 disliked record in like_monitor table then increment disliked column in comment table
+            } else { //comment has been liked
+                $comment->subtractLikedCount($comment_id); //subtract 1 from liked column in comment table
+                $comment->deleteExisting($comment_id);//deletes liked record in like_monitor table
+                $comment->addDislike($comment_id); //add 1 dislike record in like_monitor table then increment disliked column in comment table
+            }
+        }
+
+        $this->set(get_defined_vars());
+        redirect("/comment/view?page={$_SESSION['current_page']}&thread_id={$_SESSION['thread_id']}");
+    }
+
+    public function most_liked()
+    {
+        $comments = new Comment();
+        $users = new User();
+
+        $comment = $comments->getTopComments();
+        $user = $users->getAll();
+
+        if ($comment) {
+            $current_page = max(Param::get('page'), SimplePagination::MIN_PAGE_NUM);
```

Comment:
i just removed the constant inside the library po then defined it na lang po sa core